### ISSUE TITLE
Trying to open a read-only file in a non-read-only mode yields a confusing error message

### ISSUE THREAD

**Issue by mbq** [mannau/h5#34]
`h5file`, when run on a read-only file with a default mode `a` yields
```
ERROR: H5Fopen failed in H5File constructor
```
This is a confusing message, suggesting that something is wrong with the file. IMHO h5 should either infer default mode from file permissions, or produce a more explicit message.

**Comment by mannau** [comment]
You're right, I'm still thinking about this (implicit switch to read-only vs explicit error message).

**Comment by gvdr** [comment]
Any progress?
